statements.tex

```diff
@@ -11,79 +11,96 @@
 
 \begin{quote}
 \pnum
-For a range-based for statement of the form
+\oldoldtxt{For a}\newnewtxt{The} range-based \tcode{for} statement \oldoldtxt{of the form}
 
-\begin{bnf}
-\terminal{for} \terminal{(} for-range-declaration \terminal{:} expression \terminal{)} statement
-\end{bnf}
+\begin{ncbnf}
+\terminal{for (} for-range-declaration \terminal{:} \oldoldtxt{expression}\newnewtxt{for-range-initializer} \terminal{)} statement
+\end{ncbnf}
 %
-let \grammarterm{range-init} be equivalent to the expression surrounded by parentheses
+\begin{removedblock}
+\oldoldtxt{let \grammarterm{range-init} be equivalent to the expression surrounded by parentheses}
 
 \begin{bnf}
-\terminal{(} expression \terminal{)}
+\oldoldtxt{\terminal{(} expression \terminal{)}}
 \end{bnf}
 %
-and for a range-based for statement of the form
+\oldoldtxt{and for a range-based for statement of the form}
 
 \begin{bnf}
-\terminal{for} \terminal{(} for-range-declaration \terminal{:} braced-init-list \terminal{)} statement
+\oldoldtxt{\terminal{for} \terminal{(} for-range-declaration \terminal{:} braced-init-list \terminal{)} statement}
 \end{bnf}
 %
-let \grammarterm{range-init} be equivalent to the \grammarterm{braced-init-list}.
+\oldoldtxt{let \grammarterm{range-init} be equivalent to the \grammarterm{braced-init-list}.}
+\end{removedblock}
 %
-In each case, a range-based for statement is equivalent to
+\oldoldtxt{In each case, a range-based for statement} is equivalent to
+
 \begin{removedblock}
-\begin{codeblock}
-{
-  auto && __range = range-init;
-  for ( auto __begin = begin-expr,
-             __end = end-expr;
-        __begin != __end;
-        ++__begin ) {
-    @\grammarterm{for-range-declaration}@ = *__begin;
-    @\grammarterm{statement}@
-  }
-}
-\end{codeblock}
+\begin{ncbnftab}
+\terminal{\{}\br
+\>\terminal{auto \&\&__range =} range-init \terminal{;}\br
+\>\terminal{for ( auto __begin =} begin-expr \terminal{,}\br
+\>\>\terminal{ __end =} end-expr \terminal{;}\br
```

Comment:
Yes, I did some battle with the tex sources here and lost: alignment to non-tab boundaries is not easy in this ncbnftab environment. I've since thought of protecting spaces:
```tex
\terminal{{ }{ }{ }{ }{ }__end = } ...
```
which seems to work.